File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/query/cs/querybyattribute1.cs
// <snippetquerybyattribute1>


 //  Create query using QueryByAttribute.
 QueryByAttribute querybyattribute = new QueryByAttribute("account");
 querybyattribute.ColumnSet = new ColumnSet("name", "address1_city", "emailaddress1");

 //  Attribute to query.
 querybyattribute.Attributes.AddRange("address1_city");

 //  Value of queried attribute to return.
 querybyattribute.Values.AddRange("Redmond");

 //  Query passed to service proxy.
 EntityCollection retrieved = _service.RetrieveMultiple(querybyattribute);

 System.Console.WriteLine("Query Using QueryByAttribute");
 System.Console.WriteLine("===============================");

 //  Iterate through returned collection.
 foreach (var c in retrieved.Entities)
 {
     System.Console.WriteLine("Name: " + c.Attributes["name"]);
     
     if( c.Attributes.Contains("address1_city") )
         System.Console.WriteLine("Address: " + c.Attributes["address1_city"]);

     if( c.Attributes.Contains("emailaddress1") )
         System.Console.WriteLine("E-mail: " + c.Attributes["emailaddress1"]);
 }
 System.Console.WriteLine("===============================");

// </snippetquerybyattribute1>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/queryvb/vb/querybyattribute1.vb
' <snippetquerybyattribute1>


 '  Create query using QueryByAttribute.
 Dim querybyattribute As New QueryByAttribute("account")
 querybyattribute.ColumnSet = New ColumnSet("name", "address1_city", "emailaddress1")

 '  Attribute to query.
 querybyattribute.Attributes.AddRange("address1_city")

 '  Value of queried attribute to return.
 querybyattribute.Values.AddRange("Redmond")

 '  Query passed to service proxy.
 Dim retrieved As EntityCollection = _service.RetrieveMultiple(querybyattribute)

 Console.WriteLine("Query Using QueryByAttribute")
 Console.WriteLine("===============================")

 '  Iterate through returned collection.
 For Each c In retrieved.Entities
     Console.WriteLine("Name: " &amp; c.Attributes("name").ToString())

     If c.Attributes.Contains("address1_city") Then
         Console.WriteLine("Address: " &amp; c.Attributes("address1_city").ToString())
     End If

     If c.Attributes.Contains("emailaddress1") Then
         Console.WriteLine("E-mail: " &amp; c.Attributes("emailaddress1").ToString())
     End If
 Next c
 Console.WriteLine("===============================")

' </snippetquerybyattribute1>